src/modules/auth/controllers/het-archief.controller.spec.ts

```diff
@@ -234,12 +244,35 @@ describe('HetArchiefController', () => {
 	});
 
 	describe('login-callback', () => {
-		it('should redirect after successful login with a known user', async () => {
+		it('should redirect after successful login with a known user and succesful update in CM', async () => {
+			mockArchiefService.assertSamlResponse.mockResolvedValueOnce(ldapUser);
+			mockUsersService.getUserByIdentityId.mockReturnValue(archiefUser);
+			mockIdpService.determineUserGroup.mockReturnValueOnce(GroupId.CP_ADMIN);
+			mockIdpService.userGroupRequiresMaintainerLink.mockReturnValueOnce(true);
+			mockUsersService.updateUser.mockReturnValue(archiefUser);
+			mockCampaignMonitorService.updateNewsletterPreferences.mockReturnValueOnce({});
+
+			const result = await hetArchiefController.loginCallback(
+				mockRequest,
+				{},
+				samlResponse,
+				{}
+			);
+
+			expect(result).toEqual({
+				statusCode: HttpStatus.TEMPORARY_REDIRECT,
+				url: hetArchiefLoginUrl,
+			});
+			expect(mockUsersService.createUserWithIdp).not.toBeCalled();
+			expect(mockUsersService.updateUser).not.toBeCalled();
```

Comment:
also add this check for the other tests:
```suggestion
			expect(mockUsersService.updateUser).not.toBeCalled();
			expect(mockCampaignMonitorService.updateNewsletterPreferences).toBeCalledTimes(1);
```